field.go

```diff
@@ -2,10 +2,14 @@ package main
 
 import (
 	"fmt"
+	"math/rand"
 	"strconv"
 	"strings"
 )
 
+var zhashes [][][]uint32
+var zhashInitialized bool = false
```

Comment:
You don't need this boolean. The zero value of a slice is `nil`, so you can check if the slice is `nil` instead of comparing to this.